### DISCUSSION TITLE
PUT method on a favorite/booking button (changing a boolean)

### DISCUSSION THREAD

**Discussion by shrocket** [neuefische/questions#745] [Web] [answered]
Here is a link to the [PR](https://github.com/ohlunlacassi/hotel-tishjama/pull/24) with which we're working.

We are having a great deal of trouble figuring out how to code our `booking` button. This button should function just like a `favorite` button, and, when pressed, will lead the user to their booking/favorites page. The `booking` button should also change a boolean value (`isBooked`) in our MongoDB from `false` to `true` using the PUT method.

We have looked at various challenges, solutions, and examples. We have even asked our good friend ChatGPT, but alas and alack, we have not been able to find a suitable solution. Thus, we turn to you, Grand neuefische Hivemind, for help.

With many thanks, Team Tishjama

**Comment by mbosselmann** [comment, marked as answer]
Hi Hotel Tishjama Team,
one of the coaches of the ✨ neuefische hivemind ✨  is sending the following answer:

I found a solution for your booking feature and I think the main problem was that the `await` was missing inside of your `async` function `handleBooking`. :)

```js
 const router = useRouter();  

const {
    data: service,
    isLoading,
    error,
    mutate,
  } = useSWR(id ? `/api/services/${id}` : null);


  const handleBooking = async () => {
    const response = await fetch(
      `/api/services/${service._id}`,
      {
        method: "PUT",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ isBooked: !service.isBooked }),
      }
    );

    if (response.ok) {
      mutate();
      router.push("/MyBookings");
    }
  };

```

My changes:
- `setService` is a function that is not part useSWR. Maybe something left behind from a former `setState` function (useState)? 🤔 Therefore, I removed everything related to `setService`.
- I declared the variable `response` to use this below for `mutate()` and `router.push` and added the `await` keyword we need before a fetch inside of an `async` function 💪 
- I changed the `url` of the fetch to `/api/services/${service._id}` as the former `url` was not correct. 👀 
- inside of the body I changed the value to `{ isBooked: !service.isBooked }` to have the toggle effect you were looking for. ✨ 
- I removed the if statement that checked if the services is booked as it is not necessary anymore. 💪 
- But I added a new if statement to check of the response of the fetch is `ok` and to say what then should happen: get the new data (`mutate()`) and then redirect to the page `MyBookings` (this should be the right path or?).
- to use `router.push()` I imported `useRouter()` from `next/router` :)

I hope this does help!

**Reply by shrocket** [reply]
Dear Mareike,
Thank you, this looks great! And feels clean! And appears to solve our issue elegantly! We'll have a talk with Jessica and/or Diego about what to do, but we are very grateful you spent your valuable time helping us. Thanks some more.
-Team Tishjama
